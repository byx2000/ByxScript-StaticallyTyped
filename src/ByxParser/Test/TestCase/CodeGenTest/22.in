function void main()
{
	int i = f1(); // 0
	double x = f2(); // 1
	
	i = f3(20, 30);
	x = f4(1.57, 3.14);
	
	int ia = 100; // 2
	int ib = 200; // 3
	double da = 0.5; // 4
	double db = 1.5; // 5
	i = f3(ia, ib);
	x = f4(da, db);
}

function int f1()
{
	return 100;
}

function double f2()
{
	return 3.14;
}

function int f3(int a, int b)
{
	return a;
}

function double f4(double a, double b)
{
	return b;
}