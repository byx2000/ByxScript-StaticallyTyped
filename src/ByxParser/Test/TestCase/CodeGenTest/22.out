5
6 0
0 35
0 38
2 41
2 46

0

call 1
toi
istore 0
call 2
tod
dstore 1
iconst 30
iconst 20
call 3
toi
istore 0
dconst 3.14
dconst 1.57
call 4
tod
dstore 1
iconst 100
istore 2
iconst 200
istore 3
dconst 0.5
dstore 4
dconst 1.5
dstore 5
iload 3
iload 2
call 3
toi
istore 0
dload 5
dload 4
call 4
tod
dstore 1
ret

iconst 100
ret
ret

dconst 3.14
ret
ret

istore 0
istore 1
iload 0
ret
ret

dstore 0
dstore 1
dload 1
ret
ret